docs/en/latest/plugins/proxy-rewrite.md

```diff
@@ -65,68 +64,446 @@ admin_key=$(yq '.deployment.admin.admin_key[0].key' conf/config.yaml | sed 's/"/
 
-To remove the `proxy-rewrite` Plugin, you can delete the corresponding JSON configuration from the Plugin configuration. APISIX will automatically reload and you do not have to restart for this to take effect.
+The following example demonstrates how you can add URL parameters to the request.
 
 ```shell
-curl http://127.0.0.1:9180/apisix/admin/routes/1  -H "X-API-KEY: $admin_key" -X PUT -d '
+curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT \
+  -H "X-API-KEY: ${admin_key}" \
+  -d '{
+    "id": "proxy-rewrite-route",
+    "methods": ["GET"],
+    "uri": "/get",
+    "plugins": {
+      "proxy-rewrite": {
+        "uri": "/get?arg1=apisix&arg2=plugin"
+      }
+    },
+    "upstream": {
+      "type": "roundrobin",
+      "nodes": {
+        "httpbin.org:80": 1
+      }
+    }
+  }'
+```
+
+Send a request to verify if the URL parameters are also forwarded to upstream:
+
+```shell
+curl "http://127.0.0.1:9080/get"
+```
+
+You should see a response similar to the following:
+
+```text
 {
+  "args": {
+    "arg1": "apisix",
+    "arg2": "plugin"
+  },
+  "headers": {
+    "Accept": "*/*",
+    "Host": "127.0.0.1",
+    "User-Agent": "curl/8.2.1",
+    "X-Amzn-Trace-Id": "Root=1-64fef6dc-2b0e09591db7353a275cdae4",
+    "X-Forwarded-Host": "127.0.0.1"
+  },
+  "origin": "127.0.0.1, 103.248.35.148",
+  # highlight-next-line
+  "url": "http://127.0.0.1/get?arg1=apisix&arg2=plugin"
+}
+```
+
+### Rewrite HTTP Method
+
+The following example demonstrates how you can rewrite a GET request into a POST request.
+
+```shell
+curl "http://127.0.0.1:9180/apisix/admin/routes" -X PUT \
+  -H "X-API-KEY: ${admin_key}" \
+  -d '{
+    "id": "proxy-rewrite-route",
     "methods": ["GET"],
-    "uri": "/test/index.html",
-    "plugins": {},
+    "uri": "/get",
+    "plugins": {
+      "proxy-rewrite": {
+        "uri": "/anything",
+        "method":"POST"
+      }
+    },
     "upstream": {
-        "type": "roundrobin",
-        "nodes": {
-            "127.0.0.1:80": 1
+      "type": "roundrobin",
+      "nodes": {
+        "httpbin.org:80": 1
+      }
+    }
+  }'
+```
+
+Send a GET request to `/get` to verify if it is transformed into a POST request to `/anything`:
+
+```shell
+curl "http://127.0.0.1:9080/get"
+```
+
+You should see a response similar to the following:
+
+```text
+{
+  "args": {},
+  "data": "",
+  "files": {},
+  "form": {},
+  "headers": {
+    "Accept": "*/*",
+    "Host": "127.0.0.1",
+    "User-Agent": "curl/8.2.1",
+    "X-Amzn-Trace-Id": "Root=1-64fef7de-0c63387645353998196317f2",
+    "X-Forwarded-Host": "127.0.0.1"
+  },
+  "json": null,
+  "method": "POST",
+  "origin": "::1, 103.248.35.179",
+  "url": "http://localhost/anything"
+}
+```
+
+### Forward Consumer Names to Upstream
+
+The following example demonstrates how you can forward the name of consumers who authenticates successfully to Upstream services. As an example, you will be using `key-auth` as the authentication method.
+
+Create a consumer `JohnDoe`:
```

Comment:
```suggestion
Create a Consumer `JohnDoe`:
```

Reply:
Check accordingly

Reply:
this one's right; I just checked for other instances as well